Source/Scene/ModelInstanceCollection.js

```diff
@@ -0,0 +1,675 @@
+        } else {
+            vertexBuffer.copyFromArrayView(typedArray);
+        }
+    }
+
+    function updateBoundingSphere(collection) {
+        var points = [];
+        var instancesLength = collection.instancesLength;
+        for (var i = 0; i < instancesLength; i++) {
+            var translation = new Cartesian3();
+            Matrix4.getTranslation(collection._instances[i].modelMatrix, translation);
+            points.push(translation);
+        }
+
+        var boundingSphere = collection._boundingSphere;
+        BoundingSphere.fromPoints(points, boundingSphere);
+        Matrix4.fromTranslation(boundingSphere.center, collection._boundingSphereModel);
+    }
+
+    function createModel(collection, context) {
+        var i;
+        var instancingSupported = collection._instancingSupported;
+        var modelOptions = {
+            url : collection._url,
+            cacheKey : collection._cacheKey,
+            headers : collection._headers,
+            asynchronous : collection._asynchronous,
+            debugShowBoundingVolume : collection._debugShowBoundingVolume,
+            debugWireframe : collection._debugWireframe,
+            allowPicking : collection._allowPicking,
+            precreatedAttributes : undefined,
+            vertexShaderLoaded : undefined,
+            fragmentShaderLoaded : undefined,
+            uniformMapLoaded : undefined,
+            pickVertexShaderLoaded : undefined,
+            pickFragmentShaderLoaded : undefined,
+            pickUniformMapLoaded : undefined,
+            ignoreCommands : false
+        };
+
+        updateBoundingSphere(collection);
+
+        if (instancingSupported) {
+            updateVertexBuffer(collection, context);
+
+            var attributeNames = ['czm_modelMatrixRow0', 'czm_modelMatrixRow1', 'czm_modelMatrixRow2', 'czm_instanceData'];
+            var componentSizeInBytes = ComponentDatatype.getSizeInBytes(ComponentDatatype.FLOAT);
+            var instancedAttributes = {};
+
+            for (i = 0; i < 4; ++i) {
+                instancedAttributes[attributeNames[i]] = {
+                    index                  : 0, // updated in Model
+                    vertexBuffer           : collection._vertexBuffer,
+                    componentsPerAttribute : 4,
+                    componentDatatype      : ComponentDatatype.FLOAT,
+                    normalize              : false,
+                    offsetInBytes          : componentSizeInBytes * 4 * i,
+                    strideInBytes          : componentSizeInBytes * 16,
+                    instanceDivisor        : 1
+                };
+            }
+
+            collection._instancedAttributes = instancedAttributes;
+
+            // Instanced models will create different renderer resources, so change the cache key.
+            var url = collection._url;
+            var cacheKey = defaultValue(collection._cacheKey, Model.getDefaultCacheKey(url));
+            cacheKey += '(instanced)';
```

Comment:
Perhaps `/instanced` or `#instanced`.  Just something to avoid collisions (which are unlikely anyway).